ViAn/Filehandler/filehandler.h

```diff
@@ -15,93 +15,107 @@
 #include "project.h"
 #include "dir.h"
 #include <algorithm>
-
-
-
+#include <QJsonObject>
+#include <QFile>
+#include <QDir>
+#include <QTextStream>
+#include <QJsonDocument>
 enum WRITE_OPTION{APPEND, OVERWRITE};
 typedef int FH_ERROR; // file handler error code
 typedef int ID;
 struct Project; // fix for include issue
 struct ProjFiles;
 class FileHandler
 {
+    enum SaveFormat {
+        Json, Binary
+    };
+
 public:
+
     FileHandler();
-    //
-    //  Project manipulation
-    //
-    std::string work_space;
-    void set_workspace(std::string new_work_space);
-    Project* open_project(std::string dirpath);
-    Project* create_project(std::string proj_name);
-    FH_ERROR delete_project(ID id);
+
+    ID work_space;
+    void set_work_space(std::string new_work_space);
+    QDir get_work_space();
+//  Project methods
+//  Project* open_project(ID id); // To be added
+//  Project* close_project(ID id);
+
+    Project* create_project(QString proj_name, std::string dir_path="");
+    bool delete_project(ID proj_id);
+
+
     Project* load_project(std::string full_project_path);
+    Project* load_project(std::string full_path, SaveFormat save_form);
     Project* load_project(std::string proj_name, std::string dir_path);
+
     void save_project(ID id);
     void save_project(Project* proj);
-    // Video operations
+    bool save_project(Project* proj, ID dir_id, FileHandler::SaveFormat save_format);
+
+//  Video methods
     void remove_video_from_project(ID proj_id, ID vid_id);
     ID add_video(Project* proj, std::string file_path);
-    //directory manipulation
-    //varying implementation
-    ID create_directory(std::string dir_path);
-    FH_ERROR delete_directory(ID id);
+//  Directory methods
+    ID create_directory(QString dir_path);
+    bool delete_directory(ID id);
 
-    //file manipulation
+//  File methods
```

Comment:
Grouping the functions are good but do it in the same way. First the descriptive comment then the functions and then an empty line before the next grouping. Also try not to have a lot of empty lines in one group. Some are ok if they are like subgroups.
like:
//Cars
volvo
saab

//fruits
apple
pear